spark/src/main/java/org/apache/iceberg/spark/source/PartitionKey.java

```diff
@@ -176,61 +70,22 @@ public int hashCode() {
     }
   }
 
-  private static Accessor<InternalRow> newAccessor(int position, boolean isOptional, Types.StructType type,
-                                                   Accessor<InternalRow> accessor) {
+  @Override
+  protected Accessor<InternalRow> newAccessor(int position, boolean isOptional, Types.StructType type,
+                                              Accessor<InternalRow> accessor) {
     int size = type.fields().size();
     if (isOptional) {
       // the wrapped position handles null layers
       return new WrappedPositionAccessor(position, size, accessor);
     } else if (accessor.getClass() == PositionAccessor.class) {
       return new Position2Accessor(position, size, (PositionAccessor) accessor);
-    } else if (accessor instanceof Position2Accessor) {
+    } else if (accessor.getClass() == Position2Accessor.class) {
```

Comment:
It's quite confuse for me to see that the above else-if use `accessor.getClass() == PositionAccessor.class` while this line use `accessor instanceof Position2Accessor`.  I see that there's only one Position2Accessor implementation, so changing it from `instanceof`  to `=` should be OK here.

Reply:
Yes, this should be fine and is also more correct. Originally, both used `instanceof`, but there was a bug when the behavior of `PositionAccessor` was overridden. Replacing a `PositionAccessor` subclass with a `Position2Accessor` removed the subclass's behavior. It is only safe to replace a `PositionAccessor`, not a subclass.